modules/ROOT/pages/mule-high-availability-ha-clusters.adoc

```diff
@@ -18,15 +24,15 @@ Contact your customer service representative about pricing for this feature.
 
 == The Benefits of Clustering
 
-By default, clustering Mule runtime engines ensures high system availability. If a Mule runtime engine node becomes unavailable due to failure or planned downtime, another node in the cluster can assume the workload and continue to process existing events and messages. The following figure illustrates the processing of incoming messages by a cluster of two nodes. Notice that the processing load is balanced across nodes: Node 1 processes message 1 while Node 2 simultaneously processes message 2.
+By default, clustering Mule runtime engines ensures high system availability. If a Mule runtime engine node becomes unavailable due to failure or planned downtime, another node in the cluster can assume the workload and continue to process messages from VM queues as well as continue to service other requests. The following figure illustrates the processing of incoming messages by a cluster of two nodes. Notice that the processing load is balanced across nodes: Node 1 processes message 1 while Node 2 simultaneously processes message 2.
 
 image::failovernofail.png[FailoverNoFail]
 
-If one node fails, the other available nodes pick up the work of the failing node. As shown in the following figure, if Node 2 fails, Node 1 processes both message 1 and message 2.
+If one node fails, the other available nodes pick up the work of the failing node. This would be via an external load balancer redirecting the share of traffic to the still active node, or via use of VM queues to enable the active nodes to continue to process in-flight messages. As shown in the following figure, if Node 2 fails, Node 1 processes both message 1 and message 2.
 
 image::failovernode2fail.png[FailoverNode2Fail]
 
-Because all nodes in a cluster of Mule runtime engines process messages simultaneously, clusters can also improve performance and scalability. Compared to a single node instance, clusters can support more users or improve application performance by sharing the workload across multiple nodes or by adding nodes to the cluster.
+Because all nodes in a cluster of Mule runtime engines are "active active" and can process messages simultaneously, clusters can also improve performance and scalability. Compared to a single node instance, clusters can support more users or improve application performance by sharing the workload across multiple nodes or by adding nodes to the cluster. Note that not all applications will perform better with horizontally scaling or clustering - as it depends on having work of a nature that can be shared with those additional nodes, and some applications with heavy dependency on Object Stores might suffer from the work to replicate/coordinate access to the data within the shared memory grid. 
```

Comment:
```suggestion
Because all nodes in a cluster of Mule runtime engines are "active active" and can process messages simultaneously, clusters can also improve performance and scalability. Compared to a single node instance, clusters can support more users or improve application performance by sharing the workload across multiple nodes or by adding nodes to the cluster. Note that not all applications will perform better with horizontally scaling or clustering. Performance depends on whether the work can be shared with additional nodes. Performance of some applications with a heavy dependency on an Object Store can degrade because of the work required to replicate or coordinate access to the data within the shared memory grid. 
```

Reply:
The term "Ease of sharing work" kinda feels a bit more human-ish than the wording for an app server. We're talking stuff that either _is or is not_  shared (the old Yoda "do or do not, there is no try"). There's no relative difficulty involved - it's a binary yes or no - it's local or it's shared. 

And the term instances - I reckon going simply with "an object store" or "object stores" rather than object store instances. But not really that big a deal either way I don't think it's outrageous to say instances.. Although there may be an argument to talk about "partition" perhaps, although would need engineering to clarify "object store" vs "object store instance" vs "object store partition".

Reply:
@eze210 
“Performance depends on the nature of the work to be shared with additional nodes.”